Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,711 @@
+  //  selection: The RowSelection enum value defining the selection mode (disabled/single/multiple)
+  //  hasClickSensitiveRows: true if the click event handler is bound to the data rows.
+  //  onSelectionChangedHandler: A function to be invoked when the BocList's selection changes. 
+  //                             First argument is the BocList, second argument is a flag indicating whether the callback is invoked during initialization.
+  public static InitializeList(bocList: HTMLElement, selectRowSelectorControlName: string, selectAllSelectorControlName: string, selection: number, hasClickSensitiveRows: boolean, onSelectionChangedHandler: BocList_SelectionChangedHandler): void
+  {
+    if (BocList.HasDimensions (bocList))
+    {
+      $(bocList).addClass('hasDimensions');
+      BocList.FixUpScrolling($(bocList));
+    }
+
+    var selectedRows = new BocList_SelectedRows (selection);
+    if (   selectedRows.Selection != BocList._rowSelectionUndefined
+        && selectedRows.Selection != BocList._rowSelectionDisabled)
+    {
+      selectedRows.SelectRowSelectorControls = $('input[name="' + selectRowSelectorControlName + '"]');
+      selectedRows.SelectAllSelectorControls = $('input[name="' + selectAllSelectorControlName + '"]');
+      selectedRows.OnSelectionChanged = onSelectionChangedHandler;
+
+      selectedRows.SelectRowSelectorControls.each (function (this: JQuery)
+      {
+        selectedRows.DataRowCount++;
+
+        var checkBox = this;
+
+        var tableCell = checkBox.parentNode;
+        if (tableCell.nodeName !== 'TD')
+          throw 'Unexpected element type: \'' + tableCell.nodeName + '\'';
+
+        var tableRow = tableCell.parentNode;
+        if (tableRow.nodeName !== 'TR')
+          throw 'Unexpected element type: \'' + tableRow.nodeName + '\'';
+
+        if (hasClickSensitiveRows)
+          BocList.BindRowClickEventHandler (bocList, tableRow, tableCell, this);
+    
+        if (this.checked)
+        {
+          var rowBlock = new BocList_RowBlock(tableRow, this);
+          selectedRows.Rows[this.id] = rowBlock;
+          selectedRows.Length++;
+        }
+      });
+
+      BocList.SetSelectAllRowsSelectorOnDemand (selectedRows);
+    }
+    BocList._selectedRows[bocList.id] = selectedRows;
+
+    selectedRows.OnSelectionChanged (bocList, true);
+
+    // Add diganostic metadata for web testing framework (actually: should only be rendered with IRenderingFeatures.EnableDiagnosticMetadata on)
+    $(bocList).attr('data-boclist-is-initialized', 'true');
+  }
+
+  private static BindRowClickEventHandler(bocList: HTMLElement, row: HTMLElement, cell: string, selectorControl: JQuery): void
+  {
+    $(row).click(function (evt)
+    {
+      var hasSelectionChanged = BocList.OnRowClick (evt, bocList, row, selectorControl);
+      if (hasSelectionChanged)
+      {
+        var selectedRows = BocList._selectedRows[bocList.id];
+        selectedRows!.OnSelectionChanged (bocList, false);
```

Comment:
consistency please. we should place the `!` in the same way when accessing the same members. @a-ctor do we have the bang mostly at the usage site or when getting the variable? I think it's probably best to add it after getting the variable instead of at the usage site since there could be multiple usages and we want to express that this variable isn't null to begin with.

Reply:
When getting the variable is the preferred place, since it needs less `!` when the variable is used multiple times. Additionally, if the variable assignment changes, there is not need to hunt for leftover `!` across all usages.